src/server/test-suites/controllers/stop-test-suite.js

```diff
@@ -0,0 +1,47 @@
+import { provideAuthedUser } from '~/src/server/common/helpers/auth/pre/provide-authed-user'
+import { provideTestSuite } from '~/src/server/test-suites/helpers/pre/provide-test-suite'
+import { sessionNames } from '~/src/server/common/constants/session-names'
+import { stopTest } from '~/src/server/test-suites/helpers/fetch/stop-test'
+
+const stopTestSuiteController = {
+  options: {
+    pre: [provideTestSuite, provideAuthedUser]
+  },
+  handler: async (request, h) => {
+    const authedUser = request.pre.authedUser
```

Comment:
So there is a load of auth checking here. Considering to start the test we just control it via the UI controls. 
- At the moment we just show the form and controls if you are admin or own the test to start the test
- But considering we csrf this your gonna have to be clever and logged in to post to the endpoint to start the test
- To stop the test we should the controls but also do a load of validation on if you can stop it

Personally I think the validation here is a little bit overkill and muddies the water a bit.

Might need a bit of aligning

Reply:
sure, ssops auth's it anyway so we can just let the error filter back

Reply:
removed most of the auth and left it up to the backend to decide if they're allowed.